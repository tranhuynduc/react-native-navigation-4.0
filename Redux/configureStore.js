import createSagaMiddleware from 'redux-saga';
import { createStore, applyMiddleware, compose } from 'redux';
import createReducer from './createReducer';
import rootSaga from './sagas';
export default function configureStore(initialState = {}) {
  let composeEnhancers = compose;
  const reduxSagaMonitorOptions = {};

  // If Redux Dev Tools and Saga Dev Tools Extensions are installed, enable them
  /* istanbul ignore next */

  const sagaMiddleware = createSagaMiddleware(reduxSagaMonitorOptions);

  // Create the store with two middlewares
  // 1. sagaMiddleware: Makes redux-sagas work
  // 2. routerMiddleware: Syncs the location/URL path to the state
  const middlewares = [sagaMiddleware];

  const enhancers = [applyMiddleware(...middlewares)];

  const store = createStore(
    createReducer(),
    initialState,
    composeEnhancers(...enhancers),
  );
  sagaMiddleware.run(rootSaga);
  return store;
}
