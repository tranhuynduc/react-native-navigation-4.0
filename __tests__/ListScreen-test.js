/**
 * @format
 */

import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import ConnectedListScteen, { ListScreen } from '../app/containers/ListScreen';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { List } from 'react-native-paper';
import { FlatList } from 'react-native';
import { shallow } from 'enzyme';

const mockStore = configureStore([]);
describe('List Screen', () => {
  const items = [
    {
      id: 1,
      title: 'the first note',
      description: 'The description goes here',
      address: '47 Phan Văn Hân',
      image: { uri: 'https://picsum.photos/700' },
    },
  ];
  let store;
  let instaneOf;
  let wrapper;
  const props = {
    navigation: {
      navigate: jest.fn(),
      push: jest.fn(),
      openDrawer: jest.fn(),
    },
  };

  it('should render with given state from Redux store', () => {
    store = mockStore({
      notes: [],
    });
    store.dispatch = jest.fn();
    instaneOf = renderer.create(
      <Provider store={store}>
        <ConnectedListScteen />
      </Provider>,
    );
    expect(instaneOf.toJSON()).toMatchSnapshot();
  });

  describe('list item', () => {
    wrapper = shallow(<ListScreen {...props} />);

    const flatList = wrapper.find(FlatList);
    const item = flatList.renderProp('renderItem')({ item: items });
    const listItem = item.find(List.Item);

    it('should render items corecttly', () => {
      expect(flatList.length).toBe(1);
      expect(listItem.length).toBe(items.length);
    });

    it('should retrn correct keyExtractor', () => {
      const noteItem = items[0];
      const keyExtractor = flatList.props().keyExtractor(noteItem);
      expect(keyExtractor).toEqual(noteItem.id);
    });

    it('should navigate to new screen when click on item', () => {
      listItem.props().onPress();
      expect(props.navigation.push).toHaveBeenCalled();
    });

    it('should render item icon', () => {
      const itemIcon = listItem.props().left();
      expect(itemIcon.type).toEqual(List.Icon);
    });
  });
});
