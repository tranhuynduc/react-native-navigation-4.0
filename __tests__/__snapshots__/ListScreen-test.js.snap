// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Screen should render with given state from Redux store 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderRadius": 4,
      "elevation": 1,
      "flex": 1,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 0.5,
        "width": 0,
      },
      "shadowOpacity": 0.24,
      "shadowRadius": 0.75,
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flexGrow": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "paddingHorizontal": 16,
          },
          Object {
            "paddingVertical": 16,
          },
          undefined,
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
            },
            Array [
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "fontFamily": "System",
                "fontWeight": "500",
                "textAlign": "left",
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "fontSize": 20,
                  "letterSpacing": 0.15,
                  "lineHeight": 30,
                  "marginVertical": 2,
                },
                undefined,
              ],
              Array [
                Object {
                  "fontSize": 20,
                  "letterSpacing": 0.15,
                  "lineHeight": 30,
                  "marginVertical": 2,
                },
                undefined,
              ],
            ],
          ]
        }
      >
        List Notes
      </Text>
      <RCTSafeAreaView
        emulateUnlessSupported={true}
      >
        <RCTScrollView
          data={Array []}
          disableVirtualization={false}
          getItem={[Function]}
          getItemCount={[Function]}
          horizontal={false}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          numColumns={1}
          onContentSizeChange={[Function]}
          onEndReachedThreshold={2}
          onLayout={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          stickyHeaderIndices={Array []}
          updateCellsBatchingPeriod={50}
          viewabilityConfigCallbackPairs={Array []}
          windowSize={21}
        >
          <View />
        </RCTScrollView>
      </RCTSafeAreaView>
    </View>
  </View>
</View>
`;
